### PR TITLE
Add publication page test

### PR DESCRIPTION
**Pull request by TupakAmaru13**
This PR add test for Add Publication page

### CODE REVIEW

File: test/pageobjects/AddPublication.page.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const Page = require("./Page");
+
+class AddPublicationPage extends Page {
+
+    get heading() {
+        return $("//header//h6[text()='publications']");
```
Comment (Burdishka) [inline test/pageobjects/AddPublication.page.js:6]:
```suggestion
        return $("//h6");
```
There is only one tag h6 on the page

File: test/pageobjects/AddPublication.page.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        await this.inputTitle.setValue(title);
+        await this.inputImageLink.setValue(imageLink);
+        await this.description.setValue(description);
+        await this.content.setValue(publicationContent);
+    }
+
+    async save() {
+        await this.saveButton.click();
+    }
+
```
Comment (Burdishka) [inline test/pageobjects/AddPublication.page.js:37]: Unnecessary empty line

File: test/pageobjects/AddPublication.page.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        return $("//header//h6[text()='publications']");
+    }
+    get inputTitle() {
+        return $('#title');
+    }
+
+    get inputImageLink() {
+        return $('#image');
+    }
+    get description() {
```
Comment (Burdishka) [inline test/pageobjects/AddPublication.page.js:15]: need one empty line

File: test/pageobjects/AddPublication.page.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    async fillForm(title, imageLink, description, publicationContent) {
+        await this.inputTitle.setValue(title);
+        await this.inputImageLink.setValue(imageLink);
+        await this.description.setValue(description);
+        await this.content.setValue(publicationContent);
+    }
+
+    async save() {
```
Comment (Burdishka) [inline test/pageobjects/AddPublication.page.js:34]: method name should be self-explanatory (clear)
example - clickSaveButton()

File: test/pageobjects/PublicationsPage.js [hunk 5]
```diff
@@ -2,20 +2,29 @@ const Page = require("./Page");
 
 class PublicationsPage extends Page{
     get h6(){
-        return $('h6');
+        return $('//h6');
     }
-    get textAddPublication(){
+
+    get buttonAddPublication(){
```
Comment (Burdishka) [inline test/pageobjects/PublicationsPage.js:8]:
```suggestion
    get addPublicationButton(){
```

consistency - saveButton, AddPublicationPage, publicationList

**Review by Burdishka — APPROVED** [review]
Just an idea for improving

I would expand the description of the PR because you add not only Add Publication Page Test but other pages too

This way you can better show everyone from the team what you want to add and improve

**Comment by TupakAmaru13** [comment]
Thank you, got it
    On Wednesday, January 5, 2022, 06:43:51 AM PST, Oleksandr Burdeinyi ***@***.***> wrote:  
 
 


@Burdishka approved this pull request.

Just an idea for improving

I would expand the description of the PR because you add not only Add Publication Page Test but other pages too

This way you can better show everyone from the team what you want to add and improve

In test/pageobjects/AddPublication.page.js:
> @@ -0,0 +1,44 @@
+const Page = require("./Page");
+
+class AddPublicationPage extends Page {
+
+    get heading() {
+        return $("//header//h6[text()='publications']");
⬇️ Suggested change-        return $("//header//h6[text()='publications']");
+        return $("//h6");

There is only one tag h6 on the page

In test/pageobjects/AddPublication.page.js:
> +
+    get saveButton() {
+        return ***@***.***="submit"]');
+    }
+
+    async fillForm(title, imageLink, description, publicationContent) {
+        await this.inputTitle.setValue(title);
+        await this.inputImageLink.setValue(imageLink);
+        await this.description.setValue(description);
+        await this.content.setValue(publicationContent);
+    }
+
+    async save() {
+        await this.saveButton.click();
+    }
+

Unnecessary empty line

In test/pageobjects/AddPublication.page.js:
> +    }
+    get description() {

need one empty line

In test/pageobjects/AddPublication.page.js:
> +    get content() {
+        return $("//form//textarea");
+    }
+
+    get saveButton() {
+        return ***@***.***="submit"]');
+    }
+
+    async fillForm(title, imageLink, description, publicationContent) {
+        await this.inputTitle.setValue(title);
+        await this.inputImageLink.setValue(imageLink);
+        await this.description.setValue(description);
+        await this.content.setValue(publicationContent);
+    }
+
+    async save() {

method name should be self-explanatory (clear)
example - clickSaveButton()

In test/pageobjects/PublicationsPage.js:
>      }
-    get textAddPublication(){
+
+    get buttonAddPublication(){
⬇️ Suggested change-    get buttonAddPublication(){
+    get addPublicationButton(){

consistency - saveButton, AddPublicationPage, publicationList

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Review by panteleyeva — APPROVED** [review]

**Pull request merged**